app/core/Encryptor/lib.ts

```diff
@@ -29,7 +31,10 @@ class AesEncryptionLibrary implements EncryptionLibrary {
       password,
       salt,
       opts.params.iterations,
+      // We're using SHA512 but returning a key with length 256 bits.
+      // Truncating the output to 256 bits is intentional and considered safe.
       SHA256_DIGEST_LENGTH,
+      ShaAlgorithm.sha512,
```

Comment:
Just adding some references:

```suggestion
      // We're using SHA512 but returning a key with length 256 bits.
      // Truncating the output to 256 bits is intentional and considered safe.
      //
      // References:
      // - https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.180-4.pdf
      // - https://eprint.iacr.org/2010/548.pdf
      SHA256_DIGEST_LENGTH,
      ShaAlgorithm.sha512,
```